core.py

```
@@ -0,0 +1,431 @@
+            
+            if public_key and user_id not in self.crypto.peer_public_keys:
+                self.crypto.load_peer_public_key(user_id, public_key)
+        
+        if self.on_user_list_update:
+            self.on_user_list_update(self.online_users)
+                
+    def format_timestamp(self, iso_timestamp: str) -> str:
+        try:
+            dt = datetime.fromisoformat(iso_timestamp)
+            return dt.strftime('%H:%M:%S')
+        except:
```

Ruff (0.14.10)

Do not use bare `except`

(E722)